### PR TITLE
Fix issue where material instance name cannot be set without creating a default instance

### PR DESCRIPTION
**Pull request by kunyoungparkk**
This PR is related to [PR #8147](https://github.com/google/filament/pull/8147)

After applying PR #8147, FMaterial::createInstance has an issue where it ignores the name passed as a parameter when mDefaultMaterialInstance is not set.

By applying this PR, a material instance can be assigned a name even if mDefaultMaterialInstance is not set.

### CODE REVIEW

**Review by pixelflinger — APPROVED** [review]

**Comment by pixelflinger** [comment]
@kunyoungparkk would you be able to sign the CLA so we can merge this change? Thank you!

File: filament/src/details/Engine.cpp [hunk 1]
```diff
@@ -835,8 +835,10 @@ FMaterialInstance* FEngine::createMaterialInstance(const FMaterial* material,
     return p;
 }
 
-FMaterialInstance* FEngine::createMaterialInstance(const FMaterial* material) noexcept {
-    FMaterialInstance* p = mHeapAllocator.make<FMaterialInstance>(*this, material);
+FMaterialInstance* FEngine::createMaterialInstance(const FMaterial* material,
+                                                   const char* name) noexcept {
+  FMaterialInstance* p =
```
Comment (poweifeng) [inline filament/src/details/Engine.cpp:840]: nit: 4 spaces here

File: filament/src/details/MaterialInstance.h [hunk 2]
```diff
@@ -53,7 +53,8 @@ class FTexture;
 
 class FMaterialInstance : public MaterialInstance {
 public:
-    FMaterialInstance(FEngine& engine, FMaterial const* material) noexcept;
+  FMaterialInstance(FEngine& engine, FMaterial const* material,
```
Comment (poweifeng) [inline filament/src/details/MaterialInstance.h:56]: nit: spacing here.

**Review by poweifeng — APPROVED** [review]

File: filament/src/details/MaterialInstance.h (hunk 2 above)
Comment (kunyoungparkk) [reply filament/src/details/MaterialInstance.h]: Fixed spacing as suggested. Thanks for pointing that out!

File: filament/src/details/Engine.cpp (hunk 1 above)
Comment (kunyoungparkk) [reply filament/src/details/Engine.cpp:840]: Fixed spacing as suggested. Thanks for pointing that out!

**Review by poweifeng — APPROVED** [review]

**Comment by pixelflinger** [comment]
@kunyoungparkk windows build is failing

**Comment by kunyoungparkk** [comment]
> @kunyoungparkk windows build is failing

I’m currently investigating the issue as the build is working fine in my local Windows environment. It might take some time to identify the root cause. I appreciate your patience, and I will provide an update once the issue is resolved.

**Pull request merged**